src/fs/file.rs

```diff
@@ -330,6 +230,23 @@ impl File {
     }
 }
 
+#[async_trait::async_trait(?Send)]
+impl AsyncReadAt for File {
```

Comment:
I'm not sure how I feel about using async_trait exactly. This does introduce overhead, and I would rather that we employ a strategy with a base trait with poll_* methods and an Ext trait.

Reply:
Yeah, I agree that we shouldn't be using async_trait for this. But more generally, moving these methods into traits (async_trait or not) should be its own PR, and not happen in an "Add TCP,UDP,UDS support" PR.

Reply:
So, should I remove `async_trait`?

Reply:
Yes.